lib/Modules/FreeRdpHooks/FreeRdpHooks.psm1

```diff
@@ -24,33 +24,26 @@ Import-Module JujuHooks
 Import-Module OpenStackCommon
 
 function Get-Vcredist {
-    BEGIN {
+    Write-JujuWarning "Getting vcredist."
+    $vcredistUrl = Get-JujuCharmConfig -Scope "vcredist-url"
+    if(!$vcredistUrl) {
         $resourcesSupport = (Get-Command resource-get.exe -ErrorAction SilentlyContinue) -ne $null
```

Comment:
I'll sync this code when the other changes that I will be making (to `Get-JujuResource` and `Start-ExecuteWithRetry`) are in and we can merge this one as well.